### PR TITLE
Add coveralls support

### PR DESCRIPTION
**Pull request by cemsbr**
I forked this repo to test: https://coveralls.io/github/cemsbr/python-openflow

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 # For running doctests (during "python setup.py test")
 Sphinx >= 1.4.5
+sphinx_bootstrap_theme
```
Comment (diraol) [inline requirements-dev.txt]: @cemsbr this is already on ``requirements-docs.txt``. By the way, why is ``Sphinx`` on ``requirements-dev.txt``?

File: .travis.yml [hunk 2]
```diff
@@ -2,20 +2,21 @@ language: python
 python:
   - "3.4"
```
Comment (diraol) [inline .travis.yml]: I think we can remove 3.4 and 3.5, since we will not support them anymore. =)

File: requirements-dev.txt (hunk 1 above)
Comment (diraol) [inline requirements-dev.txt]: @cemsbr this is already on ``requirements-docs.txt``. By the way, why is ``Sphinx`` on ``requirements-dev.txt``?

File: .travis.yml [hunk 3]
```diff
@@ -2,20 +2,21 @@ language: python
...
   - "3.5"
-#  - "3.5-dev" # 3.5 development branch
-#  - "nightly" # currently points to 3.6-dev
+  - "3.6"
 # command to install dependencies
 install: 
-    - pip install -r requirements.txt
-    - pip install -r requirements-dev.txt
-    - pip install -r requirements-docs.txt
+  - pip install -r requirements.txt -r requirements-dev.txt
```
Comment (diraol) [inline .travis.yml]: maybe we will need ``requirements-docs.txt`` here too?

Comment (cemsbr) [inline .travis.yml]: _requirements-dev.txt_ should be enough for a developer (or CI) to run the tests.

File: .travis.yml (hunk 2 above)
Comment (cemsbr) [inline .travis.yml]: I intend to remove it once we break the compatibility.

File: requirements-dev.txt (hunk 1 above)
Comment (cemsbr) [inline requirements-dev.txt]: It is required to run doctests and a dev should run it.

Comment (diraol) [inline requirements-dev.txt]: Well, I stand with what I said. If the developer need to run the doctests, them it makes sense to me that ``requirements-docs.txt`` (docs related requirements) are installed, doesn't it?

Comment (cemsbr) [inline requirements-dev.txt]: I believe that we agreed several weeks ago that the purpose of requirement files are:

- requirements.txt: The app user must install it
- requirements-dev.txt: The app developer must install it (to develop, test and lint)
- requirements-doc.txt: Whoever is going to change/generate the docs should install it

For example, only code developer should install linters and only doc editors should install sphinx-autobuild.

Comment (diraol) [inline requirements-dev.txt]: Hum.... that makes sense.
But, is sphinx_bootstrap_theme required here? If so, why not sphinx_rtd_theme too (it is a requirement for bootstrap_theme)?

Comment (cemsbr) [inline requirements-dev.txt]: Yes, _sphinx_bootstrap_theme_ is required (the tests failed for me without it). As to _sphinx_rtd_theme_, I uninstalled it, ran `make clean html doctest` and all went fine. Are you sure it is required? Notice that it is commented out:

```
$ grep sphinx_rtd_theme docs/*
docs/conf.py:# html_theme = 'sphinx_rtd_theme'
```

Comment (diraol) [inline requirements-dev.txt]: Hum.... well, I thought that spinx_rtd_theme was a requirement for the bootstrap_theme. Maybe pip already solves it without our explicit "request".
So, I think that's all ok. Tks ;)

File: .travis.yml (hunk 3 above)
Comment (diraol) [inline .travis.yml]: @cemsbr just to point out, considering our discussion below.

You said that if someone want to develop the project, then he/she needs to install de `requirements-dev.txt`. And if someone wants to contribute with the documentation, then he/she needs to install the `requirements-docs.txt`.

So, despite the person choice, on travis.yml we should install both requirements, shouldn't we?

**Pull request merged**